catalog/spatial/registry/registry-report-viewer/src/main/java/org.codice.ddf.registry.report.viewer/RegistryReportViewer.java

```diff
@@ -278,19 +284,21 @@ public Response viewRegInfoHtml(@PathParam("metacardId") final String metacardId
             Map<String, Object> contactInfo = new HashMap<>();
 
             PersonNameType personName = person.getPersonName();
-            String firstName = "";
-            String middleName = "";
-            String lastName = "";
+            ArrayList<String> nameArray = new ArrayList<>();
+
+            if (personName == null) {
+                break;
+            }
             if (personName.isSetFirstName()) {
-                firstName = personName.getFirstName() + " ";
+                nameArray.add(personName.getFirstName());
```

Comment:
for all the `if(something.isSet) then add(something)` statements, could we instead add a helper method like `addNonEmptyValue(List<String> list, String value)` and just call it in all of these cases? That way we have one 'if' statement in the helper that will just check to see if the passed in value is null or empty and add it to the list if it isn't. Think that would reduce your complexity.